### PR TITLE
feat(chrome-ext): styling panel

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
image
## Related issues

- :rocket: Feature #1494

### CODE REVIEW

File: apps/chrome-devtools/src/app-devtools/localization-panel/localization-panel-pres.component.ts [hunk 1]
```diff
@@ -55,36 +58,42 @@ export class LocalizationPanelPresComponent implements OnDestroy {
...
-      startWith([])
+      { initialValue: '' }
     );
-    this.isTranslationDeactivationEnabled$ = this.connectionService.message$.pipe(
-      filter((message): message is IsTranslationDeactivationEnabledContentMessage => message.dataType === 'isTranslationDeactivationEnabled'),
-      map((message) => message.enabled),
-      shareReplay({bufferSize: 1, refCount: true})
+
+    this.filteredLocalizations = computed(() => search()
+      ? this.localizations().filter(({ key, description, tags, ref }) => [key, description, ...(tags || []), ref].some((value) => value?.toLowerCase().includes(search())))
```
Comment (fpaul-1A) [inline apps/chrome-devtools/src/app-devtools/localization-panel/localization-panel-pres.component.ts:87]: might be better to extract the signal value `search()` only once in a const

File: packages/@o3r/styling/src/devkit/styling-devkit.interface.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export interface StylingDevtoolsServiceOptions extends DevtoolsCommonOptions {
+  stylingMetadataPath: string;
+}
+
+/** Update styling variable */
+export interface UpdateStylingVariablesContentMessage extends OtterMessageContent<'updateStylingVariables'> {
+  variables: Record<string, string>;
+}
+
+export type StylingVariables = (CssVariable & { runtimeValue?: string })[];
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devkit.interface.ts:19]: missing comment

File: apps/chrome-devtools/src/app-devtools/theming-panel/common.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import type { StylingVariables } from '@o3r/styling';
+
+export type Variable = StylingVariables extends (infer T)[] ? T : never;
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/common.ts:3]: Would not be easier to also export this type in `@o3r/styling`?

File: apps/chrome-devtools/src/app-devtools/theming-panel/common.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export const varRegExp = /^var\(--([^,)]*).*\)$/;
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/common.ts:5]: You may need to handle `spaces` as well
```suggestion
export const varRegExp = /^var\(--([^, )]*).*\)$/;
```

(same comment in `searchFn` regexp)

File: apps/chrome-devtools/src/app-devtools/theming-panel/common.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+export const isRef = (variableValue: string) => varRegExp.test(variableValue);
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/common.ts:7]: missing comment to all exported properties

File: apps/chrome-devtools/src/app-devtools/theming-panel/common.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  variableName: string,
+  runtimeValueVariables?: Record<string, string | null | undefined>,
+  variables?: StylingVariables,
+  visitedVariable = new Set()
+): string | undefined => {
+  visitedVariable.add(variableName);
+  const variableMetadata = variables?.find((variable) => variable.name === variableName);
+  const variableValue = runtimeValueVariables?.[variableName] ?? variableMetadata?.runtimeValue ?? variableMetadata?.defaultValue ?? '';
+  if (isRef(variableValue)) {
+    const varName = variableValue.match(varRegExp)?.[1]!;
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/common.ts:22]: Hummm the combo of `?` and `!` is funny :D

Is it:
```
const varName = variableValue.match(varRegExp)?.[1];
```
or 
```
const varName = variableValue.match(varRegExp)![1];
```

File: apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, map, shareReplay, startWith, throttleTime } from 'rxjs/operators';
+import { ChromeExtensionConnectionService } from '../../services/connection.service';
+import { ColorPipe } from './color.pipe';
+import { resolveVariable, searchFn, type Variable } from './common';
+import { IsRefPipe } from './is-ref.pipe';
+import { MemoizePipe } from './memoize.pipe';
+import { VariableLabelPipe } from './variable-label.pipe';
+
+// TODO define it in @o3r/styling or @o3r/design
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts:16]: in `@o3r/styling` indeed

File: apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+  private readonly variables$: Observable<StylingVariables>;
+  private readonly subscription = new Subscription();
+  private readonly runtimeValues$ = this.form.controls.variables.valueChanges.pipe(startWith({}));
+  private readonly runtimeValues = toSignal(this.runtimeValues$, { initialValue: {} });
+
+  constructor(
+    connectionService: ChromeExtensionConnectionService
+  ) {
+    connectionService.sendMessage(
```
Comment (kpanot) [inline apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts:56]: As the JS run is faster than a postmessage it should not be an issue but you are requesting the information before be plugged to the service to receive it :)

File: apps/showcase/src/style/design-token.figma.json [hunk 9]
```diff
@@ -1,4 +1,5 @@
 {
+  "$schema": "../../../../packages/@o3r/design/schemas/design-token.schema.json",
```
Comment (kpanot) [inline apps/showcase/src/style/design-token.figma.json:2]:
```suggestion
  "$schema": "https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/design/schemas/design-token.schema.json",
```

File: packages/@o3r/styling/src/devkit/styling-devtools.service.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { DOCUMENT } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.service.ts:2]: Please avoid `httpClient` import

File: packages/@o3r/styling/src/devkit/styling-devtools.service.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import { inject, Injectable } from '@angular/core';
+import { firstValueFrom } from 'rxjs';
+import { CssMetadata } from '../core/index';
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.service.ts:5]:
```suggestion
import type { CssMetadata } from '../core/index';
```

File: packages/@o3r/styling/src/devkit/styling-devtools.token.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import { InjectionToken } from '@angular/core';
+import { StylingDevtoolsServiceOptions } from './styling-devkit.interface';
+
+export const OTTER_STYLING_DEVTOOLS_DEFAULT_OPTIONS: StylingDevtoolsServiceOptions = {
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.token.ts:4]: missing comment

File: packages/@o3r/styling/src/devkit/styling-devtools.module.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+  imports: [
+    HttpClientModule
+  ],
+  providers: [
+    { provide: OTTER_STYLING_DEVTOOLS_OPTIONS, useValue: OTTER_STYLING_DEVTOOLS_DEFAULT_OPTIONS },
+    StylingDevtoolsMessageService,
+    OtterStylingDevtools
+  ]
+})
+export class StylingDevtoolsModule {
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.module.ts:18]: should be added in `/docs/*-devtools.md`

File: packages/@o3r/styling/src/devkit/styling-devtools.message.service.ts [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+  for (const rule of cssRules) {
+    const ruleValue = getCSSVariableValueInCSSStyleDeclaration(variableName, rule.style);
+    if (ruleValue) {
+      return ruleValue;
+    }
+  }
+};
+
+@Injectable()
+export class StylingDevtoolsMessageService implements OnDestroy {
```
Comment (fpaul-1A) [inline packages/@o3r/styling/src/devkit/styling-devtools.message.service.ts:44]: missing doc ?

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@o3r/styling/src/devkit/styling-devkit.interface.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+}
+/** Styling variable */
+export type StylingVariable = CssVariable & { runtimeValue?: string };
+
+/** Get styling variables */
+export interface GetStylingVariableContentMessage extends OtterMessageContent<'getStylingVariable'> {
+  variables: StylingVariable[];
+}
+
+type StylingMessageContents =
```
Comment (sdo-1A) [inline packages/@o3r/styling/src/devkit/styling-devkit.interface.ts:26]: doc ?

File: packages/@o3r/styling/src/devkit/styling-devkit.interface.ts [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  | UpdateStylingVariablesContentMessage
+  | GetStylingVariableContentMessage;
+
+/** List of possible DataTypes for Styling messages */
+export type StylingMessageDataTypes = MessageDataTypes<StylingMessageContents>;
+
+/** List of all messages for Styling purpose */
```
Comment (sdo-1A) [inline packages/@o3r/styling/src/devkit/styling-devkit.interface.ts:33]:
```suggestion
/** List of all messages for Styling purposes */
```

File: packages/@o3r/styling/src/devkit/styling-devkit.interface.ts [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+  OtterMessageContent,
+  RequestMessagesContentMessage
+} from '@o3r/core';
+import { CssVariable } from '../core/index';
+
+export interface StylingDevtoolsServiceOptions extends DevtoolsCommonOptions {
+  stylingMetadataPath: string;
+}
+
+/** Update styling variable */
```
Comment (sdo-1A) [inline packages/@o3r/styling/src/devkit/styling-devkit.interface.ts:14]:
```suggestion
/** Update styling variables */
```

File: packages/@o3r/styling/src/devkit/styling-devtools.token.ts [hunk 18]
```diff
@@ -0,0 +1,15 @@
+import { InjectionToken } from '@angular/core';
+import { StylingDevtoolsServiceOptions } from './styling-devkit.interface';
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.token.ts:2]:
```suggestion
import type { StylingDevtoolsServiceOptions } from './styling-devkit.interface';
```

File: packages/@o3r/styling/src/devkit/styling-devtools.service.ts [hunk 19]
```diff
@@ -0,0 +1,16 @@
+import { DOCUMENT } from '@angular/common';
+import { inject, Injectable } from '@angular/core';
+import type { CssMetadata } from '../core/index';
+
+@Injectable()
+export class OtterStylingDevtools {
```
Comment (kpanot) [inline packages/@o3r/styling/src/devkit/styling-devtools.service.ts:6]: missing class and field comments

**Review by kpanot — DISMISSED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**